src/main/java/com/launchdarkly/client/StreamProcessor.java

```diff
@@ -171,4 +196,29 @@ int version() {
     }
 
   }
+
+  private final class HeartbeatDetector implements Runnable {
+
+    @Override
+    public void run() {
+      DateTime reconnectThresholdTime = DateTime.now().minusSeconds(DEAD_CONNECTION_INTERVAL_SECONDS);
+      // We only want to force the reconnect if the ES connection is open. If not, it's already trying to
+      // connect anyway, or this processor was shut down
+      if (lastHeartbeat.isBefore(reconnectThresholdTime) && es.getState() == ReadyState.OPEN) {
+        try {
+          logger.info("Stream stopped receiving heartbeats- reconnecting.");
+          es.close();
+          start();
+        } catch (IOException e) {
+          logger.warn("Encountered exception closing stream connection: " + e.getMessage());
+        } finally {
+          if (es.getState() == ReadyState.SHUTDOWN) {
+            start();
+          } else {
+            logger.warn("Expected ES to be in state SHUTDOWN, but it's currently in state " + es.getState().toString());
```

Comment:
This is a very unlikely scenario, but we should switch this to an error level statement.